test/connectors/Acc31ConnectorSpec.scala

```diff
@@ -52,6 +52,18 @@ class Acc31ConnectorSpec extends SpecBase {
       }
     }
 
+    "return NoAssociatedData error response when responded with no associated data " +
+      "and additionalTransactions field is set to Y" in new Setup {
+      when[Future[CashTransactionsResponse]](mockHttpClient.POST(any, any, any)(any, any, any, any))
+        .thenReturn(Future.successful(noDataResponse02))
+
+      running(app) {
```

Comment:
Better to make use of map instead of using await like below. Same for test at line no 77

      running(app) {
        connector.retrieveCashTransactions("can", LocalDate.now(), LocalDate.now()).map {
          result => result mustBe Left(NoAssociatedDataException)
        }
      }